_includes/langselect.html

```diff
@@ -1,17 +1,218 @@
+          {% endfor %}
+        </ul>
+      </div>
+    </div>
+  </div>
 </div>
+<script>
+  document.addEventListener('DOMContentLoaded', function() {
+    var dropdownButton = document.querySelector('.dropdown-button');
+    var dropdownContent = document.querySelector('.dropdown-content');
+
+    dropdownButton.addEventListener('click', function() {
+      dropdownContent.classList.toggle('show');
+    });
+
+    window.onclick = function(event) {
+      if (!event.target.matches('.dropdown-button') && !event.target.matches('.dropdown-button *')) {
+        var dropdowns = document.getElementsByClassName("dropdown-content");
+        for (var i = 0; i < dropdowns.length; i++) {
+          var openDropdown = dropdowns[i];
+          if (openDropdown.classList.contains('show')) {
+            openDropdown.classList.remove('show');
+          }
+        }
+      }
+    };
+
+    // Function to normalize language code
+    function normalizeLang(lang) {
+      return lang.toLowerCase().split('-')[0];
+    }
+
+    // Get user's language from browser
+    var userLang = navigator.language || navigator.userLanguage;
+    var normalizedUserLang = normalizeLang(userLang);
+
+    // Site configuration
+    var siteLanguages = {{ site.languages | jsonify }};
+    var activeLang = '{{ site.active_lang }}';
+    var defaultLang = '{{ site.default_lang }}';
+
+    // Get the stored language from session storage
+    var storedLang = sessionStorage.getItem('activeLang');
+
+    // Determine if redirection is needed
+    var shouldRedirect = false;
+    var newLang = '';
+
+    if (!storedLang) {
+      // If no stored language but browser language is different, use it
+      if (siteLanguages.includes(normalizedUserLang) && normalizedUserLang !== activeLang) {
+        shouldRedirect = true;
+        newLang = normalizedUserLang;
+      }
+    } else if (siteLanguages.includes(storedLang) && storedLang !== activeLang) {
+      // If there's a stored language and it's different from the active one, use it
+      shouldRedirect = true;
+      newLang = storedLang;
+    }
+
+    if (shouldRedirect) {
+      var currentUrl = window.location.pathname;
+      var newUrl = newLang === defaultLang ? currentUrl.replace(/^\/[^/]+/, '') : '/' + newLang + currentUrl.replace(/^\/[^/]+/, '');
+
+      // Prevent endless redirect loops by checking if the URL already contains the language
+      if (!window.location.pathname.startsWith('/' + newLang) && window.location.pathname !== newUrl) {
+        window.location.href = newUrl;
+      }
+    }
+
+    // Store the language when a user selects it (without persistence across sessions)
+    document.querySelectorAll('.dropdown-content a').forEach(function(link) {
+      link.addEventListener('click', function(event) {
+        var selectedLang = event.target.getAttribute('data-slang');
+        sessionStorage.setItem('activeLang', selectedLang);
+      });
+    });
+  });
+</script>
+<style>
+#langselect-container {
```

Comment:
Please put this into a separate css file. This should NOT be inline!

Reply:
In a file of its own with just this styling section, or what? All the stying?

Reply:
Yes. You should always put CSS into a separate file as it can be cached.

Reply:
Ok, so does this separate file need to have a specific name or is just putting it in `assets/css/` enough?

Reply:
Ok, I think I answered myself - I put it all in fx.css and it seems happy enough. Would that do?

Reply:
langsel.css

Reply:
Changed my mind to fw.css. See below